### ISSUE TITLE
Youtube live streaming stops after ~6 hours

### ISSUE THREAD

**Issue by patrakov** [mpv-player/mpv#7499]
### Important Information

Provide following Information:
- mpv version:
```
mpv 0.32.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
ffmpeg library versions:
   libavutil       56.31.100
   libavcodec      58.54.100
   libavformat     58.29.100
   libswscale      5.5.100
   libavfilter     7.57.100
   libswresample   3.5.100
ffmpeg version: n4.2.2
```
- Youtube-dl version: 2020.02.16
- Linux Distribution and Version: Arch Linux (rolling release)
- Source of the mpv binary: Distribution packages provided by Arch Linux
- If known which version of mpv introduced the problem: not known
- Window Manager and version: Cinnamon 4.4.8
- GPU driver and version: Mesa DRI Intel(R) Haswell Desktop, Mesa 19.3.4
- Possible screenshot or video of visual glitches: none

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

Try to reproduce your issue with --no-config first. If it isn't reproducible
with --no-config try to first find out which option or script causes your issue.

Describe the reproduction steps as precise as possible. It's very likely that
the bug you experience wasn't reproduced by the developer because the workflow
differes from your own.

### Expected behavior

I should be able to watch the whole live conference

### Actual behavior

Streaming fails after 6 hours

### Log file

Make a log file made with -v -v or --log-file=output.txt, paste it to
https://0x0.st/ or attach it to the github issue, and replace this text with a
link to it.

The issue will be closed for ignoring the issue template.

Will try to do it tomorrow, because doing so will take 6 hours, which would be impossible today because the stream will end earlier. So for now you have to use the copy-paste from the terminal, and unfortunately I was asked not to share the exact youtube link by the organizers of the stream.

[youtube-livestream-log.txt](https://github.com/mpv-player/mpv/files/4270398/youtube-livestream-log.txt)

### Sample files

Try to find a long live stream, e.g. a conference.

**Comment by wm4** [comment]
This is probably because the URL returned by youtube and youtube-dl is time-limited. That, or some sort of breakage with ffmpeg's hls demuxer.

**Comment by patrakov** [comment]
Are there any options besides just -v -v or --log-file=output.txt that would help debugging?

**Comment by bitraid** [comment]
The log indicates that the server returned `HTTP error 403 Forbidden`.

**Comment by patrakov** [comment]
Yes, and that's the core of this bug report! Browsers can stream live streams longer than 6 hours (maybe they refresh something?), mpv fails because youtube gives a 403 after 6 hours.

**Comment by bitraid** [comment]
Are you using cookies for `youtube-dl`?

**Comment by patrakov** [comment]
No

**Comment by wm4** [comment]
That's what I suspect, the URL has a unique ID attached and becomes invalid after 6 hours. In that case the question is how the URL could be updated, if that's possible at all. (Or whether you should just use --loop-playlist.)

Could be something else too, though.

**Comment by patrakov** [comment]
Would it be better to re-report this against youtube-dl?

**Comment by wm4** [comment]
Well, does it also happen with youtube-dl with no mpv involved? If you test that, make sure to record a verbose log of youtube-dl (I'm not sure how that works).

**Comment by patrakov** [comment]
I will try tomorrow. How exactly does mpv invoke youtube-dl?

**Comment by bitraid** [comment]
It turns out it's a known youtube-dl issue: https://github.com/ytdl-org/youtube-dl/issues/20516

**Comment by wm4** [comment]
No, it seems to be the same problem. mpv doesn't use youtube-dl to download meda.

**Comment by bitraid** [comment]
Yes, i meant it dosen't have to be reported there.

**Comment by wm4** [comment]
In theory it might be possible to update the URL at runtime, but the problem is how to do that in a sane and smooth way. I don't think the FFmpeg API even provides anything for this. (Just reopening the ffmpeg demuxer won't be smooth, may as well just restart high level playback.)

**Comment by paulguy** [comment]
I get this problem just leaving a youtube video paused too long.  Sometimes it'll just stall playback or the video will stall while audio continues and I can use a script bound to a key to just restart playback at the same time but sometimes the player just quits.

**Comment by patrakov** [comment]
OK, one important observation: In Chromium, after 6 hours of the stream, there is also a momentary artifact with the round spinner spinning instead of the video. So, aiming at getting a 100% seamless stream may be too high.

**Comment by patrakov** [comment]
I have recorded the whole thing with wireshark and instructed both mpv and Chromium to export ssl keys. So I confirm that there is a 403 error served to Chromium, too.

**Comment by wm4** [comment]
That makes everything easier. Simplest workaround is just running with --loop-playlist.

**Comment by hassanhoot** [comment]
I have this issues too 😶

**Comment by Sunda001** [comment]
same here😓

**Comment by allanlaal** [comment]
> I get this problem just leaving a youtube video paused too long. Sometimes it'll just stall playback or the video will stall while audio continues and I can use a script bound to a key to just restart playback at the same time but sometimes the player just quits.

same here :(

@patrakov: could you change title to?:
"Youtube stream/video stops after 6 hours, even if paused (ffmpeg/youtube-dl issue)"

**Comment by kasper93** [comment]
Fixed by: https://github.com/mpv-player/mpv/pull/17879

Please retest with the latest mpv build and `--curl-enabled` flag. Most of the HTTP streaming issues are resolved now, including HTTP/{2,3} support, playlist compression, connection reuse, proxy support, chunked transfer and many more.

If the issue, or variation of it, still occurs, please comment here (or report fresh issue) with the log file with `--curl-enabled --msg-level=curl=trace`.
